app/controllers/application_modal_controller.rb

```diff
@@ -0,0 +1,29 @@
+class ApplicationModalController < ApplicationController
```

Comment:
I really can't understand how I'm breaking MVC. All controller does is sets layout for modals if necessary. And I'm using ajax to dynamically change the contents of modal. To be more accurate it changes whole modal. And with this solution it is possible to open one form as a modal or as new page without problems.